### DISCUSSION TITLE
16ch Relay ULN2803A + ESP32 Dev Kit C V2

### DISCUSSION THREAD

**Discussion by DSRocker** [arendst/Tasmota#16622] [Q&A]
Hi,
I bought a 16ch Relay (like this one: https://www.amazon.de/dp/B076F7LZYN/ref=pe_27091401_487024491_TE_item) and connectet to a ESP32 Dev Kit C V2, flashed tasmota32 firmware.bin und configured the GPIO as Relay (1-16).
It's not possible to control the relays.
What do I do wrong?!

**Comment by sfromis** [comment]
Such relay boards will often require 5V signals, while the ESP32 only can provide 3.3V. Another risk factor is that it might need a higher current than available on ESP32 data pins. Do read the detailed specs for the board you bought.

**Reply by DSRocker** [reply]
the problem is, there is no datasheet for the board. I think, when I remove the optocoupler and connect the track, it would work?!

**Reply by sfromis** [reply]
Removing the optocoupler probably makes it **less** likely to work. Do try removing the ESP32, and make connections with wires, to check if relays will click with 3.3V, or only with 5V.

The data sheet of the optocouplers should be a good place to look for electrical characteristics.

**Comment by lalo-uy** [comment]
ULN2803 datasheet say it has a on voltage of 2.4V for 200mA output.
A direct connection should work.
Any way try with wires as sugested.
